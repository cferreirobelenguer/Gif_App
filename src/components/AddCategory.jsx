import React, { useState } from 'react'

export const AddCategory=({onNewCategory})=>{
    const [inputValue,setInputValue]=useState("")

    const handlerChange=({target})=>{
        setInputValue(target.value)
    }

    const onSubmit=(event)=>{
        
        //onSubmit es llamado en la etiqueta form y en el botón submit
        //Esto significa que se mandan los datos tanto si presonamos enter con el teclado como si presionamos el botón
        event.preventDefault()
        if(inputValue.trim().length<=1) return;
        // Se coge el inputValue y se pone de parámetro en el método obtenido por props onNewCategory
        //Este método se va a encargar de modificar el arreglo dentro del state categories del componente GiftExpertApp
        onNewCategory(inputValue.trim())
        
        
    }
    return(
        <div className="card_input">
            <div className="form_submit">
                <form onSubmit={onSubmit} aria-label="form">
                    <input 
                        className="inputStyle"
                        type="text"
                        placeholder="Introduce tu gift a buscar"
                        onChange={handlerChange}
                        required    
                        value={inputValue}
                    />
                </form>
            </div>
            <div className="submit_container">
                <button onClick={onSubmit} className="btn_submit">Enviar</button>
            </div>
            
            
        </div>
        
    )
}