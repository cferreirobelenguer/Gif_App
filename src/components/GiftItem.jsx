import React from 'react'

export const GifItem=({title,url})=>{
    return(
        <div className="card">
            
            <img src={url} alt={title} className="imagenStyle"/>
            <p>{title}</p>
        </div>
    )
}