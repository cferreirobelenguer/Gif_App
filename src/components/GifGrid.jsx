import React,{useEffect} from 'react'
import { UseFetchGifs } from '../assets/hooks/useFetchGifs'
import { GifItem } from '../components'

export const GifGrid=({category,idKey})=>{
    const {datos,getGifts}=UseFetchGifs(category)

    useEffect(()=>{
        getGifts(category)
    },[category])
    
    return(
        <div>
            <div key={idKey} className='valor'>
                <h3 className='valor'>{category}</h3>
                {datos.map((image)=>{
                    return(
                        <div key={image.id} className="card-grid">
                            <GifItem image={image}></GifItem>
                        </div>
                    )
                    
                    

                })}
            </div>

        </div>
    )
}
