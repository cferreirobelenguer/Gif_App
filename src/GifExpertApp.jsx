import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp=()=>{
    const [categories,setCategories]=useState([])

    const onAddCategory=(newCategory)=>{
        //Si la categoría existe que no haga nada, y si sí existe que lo inserte en el state
        if(categories.includes(newCategory)){return;}
        setCategories([newCategory,...categories])
        console.log(newCategory)
    }

    return(
        <div className="card_container">
            <div className="card_title">
                <h1>GifExpertApp</h1>
            </div>
            <AddCategory  onNewCategory={event=>onAddCategory(event)}></AddCategory>
            <div className="card_categories">
                
                    {categories.map(i=>(
                            <div key={i} >
                                <GifGrid 
                                    category={i}
                                    idKey={i}>
                                </GifGrid>
                            </div>
                        )
                    )}
            </div>
        </div>
    )
}