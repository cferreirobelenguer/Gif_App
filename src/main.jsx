import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifExpertApp } from './GifExpertApp'
import '../src/assets/scss/style.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp></GifExpertApp>
  </React.StrictMode>
)
