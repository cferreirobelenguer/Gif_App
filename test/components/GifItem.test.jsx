import { GifItem } from '../../src/components';
const { render, screen } = require("@testing-library/react")


const title="Animation Lol GIF by Disney"
const url="https://giphy.com/gifs/disney-animation-bambi-IgLt8tZcGGYXS"

describe("Pruebas en <GifItem/>",()=>{
    test("Debe de hacer match con el snapshot",()=>{
        const {container}=render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    })
})

test("debe de mostrar la imagen con el URL y el ALT indicado",()=>{
    render(<GifItem title={title} url={url}/>)
    screen.debug()
})