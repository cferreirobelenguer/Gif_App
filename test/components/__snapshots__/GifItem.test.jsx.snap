// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Animation Lol GIF by Disney"
      class="imagenStyle"
      src="https://giphy.com/gifs/disney-animation-bambi-IgLt8tZcGGYXS"
    />
    <p>
      Animation Lol GIF by Disney
    </p>
  </div>
</div>
`;
